src/time_travel/patchers/tests/test_select.py

```diff
@@ -0,0 +1,128 @@
+        
+    def test_adding_unwaited_events(self):
+        first_event = mock.MagicMock(name='first_event')
+        second_event = mock.MagicMock(name='second_event')
+        waited_for_event = mock.MagicMock(name='waited_for_event')
+        
+        self.events_pool.add_future_event(3, first_event)
+        self.events_pool.add_future_event(4, second_event)
+        self.events_pool.add_future_event(5, waited_for_event)
+        
+        assert select.select([waited_for_event], [], [], 7) == \
+            ([waited_for_event], [], [])
+        
+        assert self.clock.timestamp == 5
+        
+    def test_multiple_events_for_same_time(self):
+        event1 = mock.MagicMock(name='event1')
+        event2 = mock.MagicMock(name='event2')
+        event3 = mock.MagicMock(name='event3')
+        event4 = mock.MagicMock(name='event4')
+        unwaited_event = mock.MagicMock(name='unwaited_event')
+        
+        self.events_pool.add_future_event(3, event1)
+        self.events_pool.add_future_event(3, event2)
+        self.events_pool.add_future_event(3, event3)
+        self.events_pool.add_future_event(3, event4)
+        self.events_pool.add_future_event(3, unwaited_event)
+        
+        returned_events = select.select(
+            [event1, event2],
+            [event3],
+            [event4],
+            7
+        )
+        
+        expected_events = ([event1, event2], [event3], [event4])
+        
+        for returned, expected in zip(returned_events, expected_events):
+            assert set(returned) == set(expected)
+        
+        assert self.clock.timestamp == 3
+
+    def test_event_not_returned_twice(self):
+        event = mock.MagicMock()
+        
+        self.events_pool.add_future_event(3, event)
+        
+        assert select.select([], [], [event], 6) == ([], [], [event])
+        assert self.clock.timestamp == 3
+        
+        assert select.select([], [], [event], 6) == ([], [], [])
+        assert self.clock.timestamp == 3 + 6
+          
+    def test_same_event_multiple_timestamps(self):
+        event = mock.MagicMock()
+        
+        self.events_pool.add_future_event(1, event)
+        self.events_pool.add_future_event(2, event)
+        
+        assert select.select([], [], [event], 6) == ([], [], [event])
+        assert self.clock.timestamp == 1
+        
+        assert select.select([event], [], [event], 6) == ([event], [], [event])
+        assert self.clock.timestamp == 2
+
+    def test_select_with_no_timeout(self):
+        event = mock.MagicMock()
+        
+        self.events_pool.add_future_event(3, event)
+        
+        assert select.select([event], [], []) == ([event], [], [])
+        assert self.clock.timestamp == 3
+
+    def test_select_infinite_wait(self):
+        event = mock.MagicMock()
+        
+        with pytest.raises(ValueError):
+            select.select([event], [], [])
+            
+    def test_event_returned_in_multiple_lists(self):
```

Comment:
That is not a good test, and it raises an issue. You have to define in your event pool event types. 
Usually when you have a single socket in both the r_list and w_list is when you buffer data in an incoming and outgoing FIFOs. In that case, if the OUT_FIFO has data and the socket is write ready - you'd read data from the fifo and send it to the socket. If the socket is read ready, you'd read from it and put the data in the IN_FIFO.

If you don't allow users to chose which event has happened in their socket, they wouldn't be able to test specific cases in their UT.